<launch>

    <node pkg="coax_server"
          name="coax"
          type="coax_server"
	  output="screen"
          args="/dev/ttyS0:1">
      <remap from="/coax_server/state" to="/state"/>

      <remap from="/coax_server/control" to="/control"/>
      <remap from="/coax_server/rawcontrol" to="/rawcontrol"/>

      <remap from="/coax_server/reach_nav_state" to="/reach_nav_state"/>
      <remap from="/coax_server/configure_comm" to="/configure_comm"/>
      <remap from="/coax_server/set_timeout" to="/set_timeout"/>
    </node>

    <node pkg="coax_vision"
          name="CoaxVisionControl"
          type="CoaxVisionControl"
	  output="screen">
      <remap from="/coax_vision/odom" to="/odom_smooth"/>
      <remap from="/coax_vision/state" to="/state"/>
      <remap from="/coax_vision/fmdes" to="/fmdes"/>
      <remap from="/coax_vision/nav_mode" to="/matlab_nav_mode"/>

      <remap from="/coax_vision/rawcontrol" to="/rawcontrol"/>
      <remap from="/coax_vision/info" to="/coax_info"/>
      <remap from="/coax_vision/imu" to="/coax_imu"/>
      <remap from="/coax_vision/control_mode" to="/control_mode"/>

      <remap from="/coax_vision/reach_nav_state" to="/reach_nav_state"/>
      <remap from="/coax_vision/configure_comm" to="/configure_comm"/>
      <remap from="/coax_vision/set_timeout" to="/set_timeout"/>
      <remap from="/coax_vision/set_control_mode" to="/set_control_mode"/>

      <param name="frequency" value="100"/>
      <param name="CoaX" value="56"/>
<!--
      <rosparam file="$(find coax_vision)/config/coax_parameters.yaml"/>
      <rosparam file="$(find coax_vision)/config/coax_control_params.yaml"/>
-->
    </node>

</launch>
